src/frontend/src/optimizer/plan_node/logical_agg.rs

```diff
@@ -208,8 +198,15 @@ impl LogicalAgg {
                 ),
                 None,
             );
-            Ok(global_agg.into())
-        }
+            global_agg.into()
+        };
+        Self::add_row_merge_if_needed(
+            needs_row_merge,
+            approx_percentile,
+            global_agg,
+            approx_percentile_col_mapping,
+            non_approx_percentile_col_mapping,
+        )
```

Comment:
Can just reuse the logic via the refactored functions above, from two phase stateless agg.